Hi, haven't had much time to test this extensively, but basic operations seem to work good :)

I've, by mistake, stumbled upon situation where I wanted to add macro to non existing template and got this error:
```bash
Traceback (most recent call last):                                                                                                                                                                                                                                               File "/tmp/user/0/ansible_z2zydb/ansible_module_zabbix_template.py", line 523, in <module>
    main()
  File "/tmp/user/0/ansible_z2zydb/ansible_module_zabbix_template.py", line 501, in main
    if existing_template_json:
UnboundLocalError: local variable 'existing_template_json' referenced before assignment
```

Maybe some kind of check would be appropriate and then `module.fail_json(msg='Modifying non-existing template')` instead of raw traceback. 


This may be a bit out of scope for module's EXAMPLES section, but I would like to see very minimal template_json example as well. Some kind of JSON struct introducing maybe one item and one trigger? :) 

I would suggest to at least add link to [Zabbix documentation](https://www.zabbix.com/documentation/3.0/manual/api/reference/template/create), but in this case it is not very helpful for building up an appropriate JSON struct.

--- a/lib/ansible/modules/monitoring/zabbix/zabbix_template.py
+++ b/lib/ansible/modules/monitoring/zabbix/zabbix_template.py
@@ -459,6 +459,7 @@ def main():
 
     template = Template(module, zbx)
     template_ids = template.get_template_ids([template_name])
+    existing_template_json = None
     if template_ids:
         existing_template_json = template.dump_template(template_ids)
 
@@ -491,6 +492,7 @@ def main():
         group_ids = None
         if template_groups is not None:
             # If the template exists, compare the already set groups
+            existing_groups = None
             if existing_template_json:
                 existing_groups = set(list(group['name'] for group in existing_template_json['zabbix_export']['groups']))
             if not existing_groups or set(template_groups) != existing_groups:
@@ -498,6 +500,7 @@ def main():
 
         macros = None
         if template_macros is not None:
+            existing_macros = None
             if existing_template_json:
                 existing_macros = set(existing_template_json['zabbix_export']['templates'][0]['macros'])
             if not existing_macros or set(template_macros) != existing_macros:

--- a/lib/ansible/modules/monitoring/zabbix/zabbix_template.py
+++ b/lib/ansible/modules/monitoring/zabbix/zabbix_template.py
@@ -117,6 +117,27 @@ EXAMPLES = '''
       - Webservers
     state: present
 
+# Import a template from Ansible variable dict
+- name: Import Zabbix Template
+  zabbix_template:
+    login_user: username
+    login_password: password
+    server_url: http://127.0.0.1
+    template_name: Test Template
+    template_json:
+        zabbix_export:
+        version: '3.2'
+        templates:
+          - name: Template for Testing
+            description: 'Testing template import'
+            template: Test Template
+            groups:
+              - name: Templates
+            applications:
+              - name: Test Application
+    template_groups: Templates
+    state: present
+
 # Add a macro to a template
 - name: Set a macro on the Zabbix template
   local_action:

